### PR TITLE
Document how the registration fields are defined

### PR DESCRIPTION
**Pull request by ghost**
Admins should be able to configure regcfp without having to look at the code to see what everything does. This is a start towards documenting the config.json file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,3 +21,44 @@ To run the code:
...
+
+There are also some special properties for field objects:
+
+  * `type`: see below for list of types
+  * `display_name`: name shown to the user
+  * `short_display_name`: name used when listing all registrations
+  * `required`: if this true and the field has empty value when the
+    registration form is submitted, the user will see an error asking
+    them to fill in the field.
+  * `private`: if this is true, the field will be shown to anyone viewing
```
Comment (puiterwijk) [inline README.md]: If this is what the code implements, I've done it wrong :(

Comment (ghost) [reply README.md]: oh, what should it do?

Comment (puiterwijk) [reply README.md]: Well, it should be that false is shown to anyone, and true is only visible to admins and the user themselves.

**Comment by ghost** [comment]
OK, new version of the patch which doesn't invert the meaning of 'private' :-)

**Comment by ghost** [comment]
Patch updated to correctly document utils.get_reg_fields()

**Comment by ghost** [comment]
Let's document the `placeholder` parameter too

**Pull request merged**